Fix cosmosdb pagination bug

itaiad200 opened the pull request:
closes #6346 6346

N-o-Z reviewed (changes requested):
Nothing about the code:
1. How does the bug manifests?
2. Can we link a an issue to this PR?
3. Can we add a unit test to catch this scenario?

itaiad200 commented:
> Nothing about the code:
> 
> 1. How does the bug manifests?
> 2. Can we link a an issue to this PR?
> 3. Can we add a unit test to catch this scenario?

Thanks, fixed all of the above. Added a test and tested locally against a live CosmosDB account.

N-o-Z commented on pkg/kv/kvtest/store.go:
```diff
@@ -420,6 +420,46 @@ func testStoreScan(t *testing.T, ms MakeStore) {
+			}
+			entries = append(entries, *ent)
+		}
+		if err := scan.Err(); err != nil {
+			t.Fatal("scan ended with an error", err)
+		}
```
```suggestion
		require.NoError(t, scan.Err(), "scan ended with an error")
```

N-o-Z reviewed (approved):
Thanks, approved with 2 suggestions (always prefer one liners 😄 )
